tests/rptest/tests/test_si_cache_space_leak.py

```diff
@@ -134,6 +134,17 @@ def is_cache_file(fname):
                 files_count += sum(1 for f in files if is_cache_file(f))
             return files_count == 0
 
+        # Reader should eventually trigger some SI cache reads when
+        # retention settings evict segment from local disk.
+        wait_until(lambda: not cache_files_closed(),
```

Comment:
should we explicitly cherry-pick https://github.com/redpanda-data/redpanda/commit/4de33829cbf8f8aa5885aff625c7805caf75a8d5 instead of bringing the changes manually?